website/docs/tracing.md

```diff
@@ -0,0 +1,95 @@
+- The code generation and execution phase, where the CodeGenerator generates the code and the CodeExecutor executes it.
+- The reply phase, where the Planner sends the reply to the user.
+
+The bars with a black line represent the critical path of the task execution, which is the longest path through the task execution. 
+This is useful for identifying the bottleneck of the task execution.
+We can clearly see that, currently, the task execution is dominated by the calls to the LLM.
+
+We can click the span (a unit of work in the trace) to see the details of the span, including the logs and the attributes.
+
+The screenshot below shows the prompt of the CodeGenerator to the LLM:
+
+![Tracing Prompt](../static/img/trace_prompt.png)
+
+We also recorded the generated code, the posts between different roles, etc. in the trace.
+
+There are also views of the trace, for example the call graph view, which shows the call hierarchy of the spans.
+Here is the call graph of the trace:
+
+![Tracing Call Graph](../static/img/trace_graph.png)
+
+## How to enable tracing
+
+Tracing is by default disabled. To enable tracing, you need to install packages required by OpenTelemetry.
+Please check the [OpenTelemetry website](https://opentelemetry.io/docs/languages/python/) for the installation guide.
+It basically requires you to install the `opentelemetry-api`, `opentelemetry-sdk`, `opentelemetry-exporter-otlp`, 
+and `opentelemetry-instrumentation` packages.
+After installing the packages, you can enable tracing by setting the `tracing.enabled=true` in the project configuration file.
+
+Next, you need to set up a trace collector and a frontend to collect and view the traces. We recommend using [Jaeger](https://www.jaegertracing.io/), 
+which is a popular open-source tracing system.
+To start, please visit the [Getting Started](https://www.jaegertracing.io/docs/getting-started/) page of Jaeger.
+An "All-in-one" Docker image is available, which is easy to start and use.
+This docker image includes both the OpenTelemetry collector and the Jaeger frontend.
+If the container is running at the same host as the TaskWeaver, you don't need to configure anything else.
+Otherwise, you need to set the `tracing.endpoint` in the project configuration file to the endpoint of the OpenTelemetry collector.
+The default endpoint of the OpenTelemetry collector is `http://127.0.0.1:4318/v1/traces`.
+
+After running the docker image, you can access the Jaeger frontend at `http://localhost:16686`.
+Now, when you run TaskWeaver, issue a task, and access the Jaeger frontend, you can see the traces of the task execution.
+On the left side panel, you can select the Service dropdown to filter the traces by the service name.
+The service name of TaskWeaver is `taskweaver.opentelemetry.tracer`.
+
+## How to customize tracing
+
+The instrumentation of TaskWeaver is done by the OpenTelemetry Python SDK.
+So, if you want to customize the tracing, you need to modify the TaskWeaver code.
```

Comment:
Can users add tracing to the plugin? guess they do not have the needs to instrument TaskWeaver core, but for plugins, it is highly possible

Reply:
It is a bit complicated to consider the generated code in tracing. I think it possible, but we do not quite understand the motivation. 

The current tracing is to let users know the time consumption of each part and potentially the costs.

Reply:
My thought was: plugin execution might take some time... we can address this later if someone has the needs.